demos/asp-net-core/Divergent.Website/Views/Orders/Index.cshtml

```diff
@@ -11,8 +11,11 @@
 
         @foreach (dynamic order in Model.Orders)
         {
+            long orderNumber = order.OrderNumber;
+            string orderNumberLink = $"Number: #{order.OrderNumber}";
+
             <tr>
-                <td><a href="/Orders/Details/@order.OrderNumber">Number: #@order.OrderNumber</a></td>
+                <td><a asp-area="" asp-controller="Orders" asp-action="Details" asp-route-id=@orderNumber>@orderNumberLink</a>
```

Comment:
Do the two new variables add enough value vs the single line option?

```html
<td><a asp-controller="Orders" asp-action="Details" asp-route-id=@order.OrderNumber>Number: #@order.OrderNumber</a>
```

Reply:
I'm on the fence, I like to have them split out for clarity, it's more readable IMO, but that portion of the code is not the main point of the demo, so we can change it as you suggest. No problems.

Reply:
One thing I definitely prefer is having `"Number: ..."` inside the `<a>` tag, otherwise the content is pulled away from the markup that surrounds it, purely because this is a templated file.

Assuming you agree, that would remove the need for the `orderNumberLink` variable, leaving only `orderNumber`. If you feel you'd rather have `@orderNumber` in the markup than `@order.OrderNumber` then I'd be OK with that.

Reply:
Code updated to match:

```
<td><a asp-controller="Orders" asp-action="Details" asp-route-id=@order.OrderNumber>Number: #@order.OrderNumber</a>
```